tests/driver_lis2dh12/main.c

```diff
@@ -11,170 +11,699 @@
+    }
+
+    /* read acceleration data */
+    else if (strncmp(argv[1], "read", sizeof("read")) == 0) {
+        uint8_t amount = (argc < 3) ? 1 : atoi(argv[2]);
+        uint8_t amt = 0;
 
         /* read sensor data */
-        int16_t data[3];
-        if (lis2dh12_read(&dev, data) != LIS2DH12_OK) {
-            puts("error: unable to retrieve data from sensor");
-            continue;
+        for (amt = 0; amt < amount; amt++){
+            if (lis2dh12_read(&dev, data_lis) != LIS2DH12_OK) {
+                puts("error: no data from sensor");
+                return -1;
+            }
+            /* format data */
+            for (int i = 0; i < 3; i++) {
+                size_t len = fmt_s16_dfp(str_out[i], data_lis[i], -3);
+                str_out[i][len] = '\0';
+            }
+
+            /* print data to STDIO */
+            printf("X: %6s  Y: %6s  Z: %6s\n", str_out[0], str_out[1], str_out[2]);
+
+            xtimer_msleep(250);
+        }
+        return 1;
+    }
+    else if (strncmp(argv[1], "read_fifo", sizeof("read_fifo")) == 0) {
+        uint8_t number = 0;
+        if ((argc < 3) || (number = atoi(argv[2])) > 32) {
+            puts("Error: Missing parameter.");
+            puts("The command should contain number of FIFO values to read (max. 32)).");
+            puts("USAGE: lis read_fifo [number]");
+            return -1;
         }
 
-        /* format data */
-        for (int i = 0; i < 3; i++) {
-            size_t len = fmt_s16_dfp(str_out[i], data[i], -3);
-            str_out[i][len] = '\0';
+        /* read raw data from FIFO */
+        uint8_t number_read = lis2dh12_read_fifo_data(&dev, data_fifo, number);
+
+        DEBUG("[lis_command]: fifo_read %d elements.\n", number_read);
+
+        /* print data */
+        for (int entry = 0; entry < number_read; entry++){
+
+            /* format data */
+            size_t len = fmt_s16_dfp(str_out[0], data_fifo[entry].X_AXIS, -3);
+            str_out[0][len] = '\0';
+            len = fmt_s16_dfp(str_out[1], data_fifo[entry].Y_AXIS, -3);
+            str_out[1][len] = '\0';
+            len = fmt_s16_dfp(str_out[2], data_fifo[entry].Z_AXIS, -3);
+            str_out[2][len] = '\0';
+
+            printf("[%2d] X: %6s  Y: %6s  Z: %6s\n", entry, str_out[0], str_out[1], str_out[2]);
         }
+        return 1;
+    }
+
+    /* clear memory */
+    else if (strncmp(argv[1], "clear_data", sizeof("clear_data")) == 0) {
+
+        lis2dh12_clear_data(&dev);
+        return 1;
+    }
+
+#ifdef MODULE_LIS2DH12_INT
+    /* set commands */
+    else if (strncmp(argv[1], "set-click", sizeof("set-click")) == 0) {
+        uint8_t thold = 0;
+        if ((argc < 3) || (thold = atoi(argv[2])) > 127) {
+            puts("Error: Missing parameter.");
+            puts("The command should contain a threshold value below 128. "
+                 "The LSB changes according to selected SCALE "
+                 "(@2G LSB=16mg; @4G LSB=32mg; @8G LSB=62mg: @16G LSB=186mg).");
```

Comment:
Isn't this contradicting itself?
186 is not below 128.

Reply:
Only the first Bit (LSB) equals to the defined `mg` value. So for setting a specific acceleration, the value should be devided by the LSB value corresponding to the current used SCALE. Maybe the driver should handle this and the user only needs to define the acceleration in `mg`. Then also a check of the current SCALE is necessary.

Reply:
> Maybe the driver should handle this and the user only needs to define the acceleration in mg

Yes that would be the proper way to do it.
If the driver only exposes raw register values to the user, it is not very helpful.